src/workerd/api/streams/standard.c++

```diff
@@ -983,15 +970,11 @@ void ReadableImpl<Self>::pullIfNeeded(jsg::Lock& js, jsg::Ref<Self> self) {
 
   auto onFailure = JSG_VISITABLE_LAMBDA((this, self = self.addRef()), (self),
                            (jsg::Lock& js, jsg::Value reason) {
-    algorithms.pulling = nullptr;
+    pulling = false;
     doError(js, kj::mv(reason));
   });
 
-  algorithms.pulling = maybeRunAlgorithm(js,
-                                         algorithms.pull,
-                                         kj::mv(onSuccess),
-                                         kj::mv(onFailure),
-                                         self.addRef());
+  maybeRunAlgorithm(js, algorithms.pull, kj::mv(onSuccess), kj::mv(onFailure), self.addRef());
```

Comment:
what was the reason to store the resulting promise before? Won't this abandon and cancel it now?

Reply:
JS promises do not get canceled the way kj promises do. Holding these before was defensive only but it's not necessary.